docs(security): extend security policy (#1142)
orhun/git-cliff: SECURITY.md

> # Security Reporting

Reviewer: Should we maybe keep the old policy and add a new section as "Reporting" instead? I kinda find the previous page useful in terms of letting people know which versions of the project will support security updates.

Author: Agree 100%, and it is still in, look in the diff under "## Supported Versions"

| 1.0.x   | :x:                |
| < 0.1.0 | :x:                |

## Reporting

To report a security vulnerability, please provide the following information:

1. **PROJECT**

>    - Include the URL of the project repository - Example: <https://github.com/orhun/git-cliff>

Reviewer: This is fixed for this repo and probably won't change, so maybe remove this step?

2. **PUBLIC**
   - Indicate whether this vulnerability has already been publicly discussed or disclosed.
   - If so, provide relevant links.

3. **DESCRIPTION**
   - Provide a detailed description of the security vulnerability.
   - Include as much information as possible to help us understand and address the issue.

> Send this information, along with any additional relevant details, to <email AT somewhere or other channel>.

Reviewer: ```suggestion
Send this information, along with any additional relevant details, to <orhunparmaksiz@gmail.com>
```

## Confidentiality

We kindly ask you to keep the report confidential until a public announcement is made.

## Notes

- Vulnerabilities will be handled on a best-effort basis.
- You may request an advance copy of the patched release, but we cannot guarantee early access before the public release.
- You will be notified via email simultaneously with the public announcement.
- We will respond within a few weeks to confirm whether your report has been accepted or rejected.

> Thank you for helping to improve the security of our project!

Reviewer: ```suggestion
Thank you for helping to improve the security of **git-cliff**!
```
